src/plugins/DevTools/DevTools.ts

```diff
@@ -4,9 +4,25 @@ import devtoolsTemplates from './DevTools.xml';
 import { OwlUIComponent } from '../../ui/OwlUIComponent';
 
 export class DevTools extends JWOwlUIPlugin {
-    constructor() {
-        super();
-        this.componentsRegistry.push(DevToolsComponent as typeof OwlUIComponent);
-        this.templates = devtoolsTemplates;
+    static templates = devtoolsTemplates;
+    Components = [DevToolsComponent as typeof OwlUIComponent];
+    constructor(dispatcher, env) {
+        super(dispatcher, env);
+        this._mountAndRegisterComponents(); // called instead of init
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Mount and register all components.
+     */
+    _mountAndRegisterComponents(): void {
+        const components = this._instantiateComponents(this.Components);
+        components.forEach((component: OwlUIComponent<{}, {}>): void => {
```

Comment:
I don't think you have to specify types in forEach. I personally thinks it makes the code very hard to read. Is `components` an array of `OwlUIComponent` or an array of functions with `(component: OwlUIComponent): void` as signature ? I know the answer but just by reading the code ? Took me a while to figure what was going on.